pkg/modelregistry/modelregistry.go

```diff
@@ -26,10 +27,19 @@ import (
 	"strings"
 )
 
+// ReadOnlySubPathMap abstracts the read only subpath
+type ReadOnlySubPathMap map[string]change.ValueType
+
+// ReadOnlyPathMap abstracts the read only path
+type ReadOnlyPathMap map[string]ReadOnlySubPathMap
+
+// GlobalReadOnlyPaths abstracts the map for the read only paths
+type GlobalReadOnlyPaths map[string]ReadOnlyPathMap
```

Comment:
I was wondering if this really needs its own type - it's only used once

Reply:
we do that here and there, I think for easiness of read it can be, and it might also be used in other places.